src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -582,20 +593,19 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType)
             return result;
         }
 
-        private LambdaSymbol CreateLambdaSymbol(NamedTypeSymbol delegateType, Symbol containingSymbol, out MethodSymbol invokeMethod, out TypeWithAnnotations returnType, out DiagnosticBag diagnostics, out ReturnInferenceCacheKey cacheKey)
+        private LambdaSymbol CreateLambdaSymbol(NamedTypeSymbol delegateType, Symbol containingSymbol, out MethodSymbol invokeMethod, out TypeWithAnnotations returnType, out BindingDiagnosticBag diagnostics, out ReturnInferenceCacheKey cacheKey)
```

Comment:
>out BindingDiagnosticBag diagnostics [](start = 170, length = 36)

Why even bother having this parameter anymore? It's just going to return an empty bag.

Reply:
> Why even bother having this parameter anymore? It's just going to return an empty bag.

The callers are relying on this function to allocate the bag. I see no good reason to change that in general and especially in context of this big refactoring.

---
In reply to: [361726420](https://github.com/dotnet/roslyn/pull/40560#discussion_r361726420) [](ancestors = 361726420)